app/scripts/modules/core/src/pipeline/config/parameters/Parameter.tsx

```diff
@@ -0,0 +1,226 @@
+import * as React from 'react';
+
+import { HelpField } from 'core/help';
+import { Tooltip } from 'core/presentation';
+import { StageConfigField } from 'core/pipeline';
+import { SortableHandle } from 'react-sortable-hoc';
+
+export interface IParameter {
+  name: string;
+  label: string;
+  required: boolean;
+  pinned: boolean;
+  description: string;
+  default: string;
+  hasOptions: boolean;
+  options: IParameterOption[];
+}
+
+export interface IParameterOption {
+  value: string;
+}
+
+export interface IParameterProps extends IParameter {
+  isMultiple?: boolean;
+  removeParameter: () => void;
+  setPinAllParametersState: () => void;
+  updateParameterField: (changes: { [key: string]: any }) => void;
+}
+
+export class Parameter extends React.Component<IParameterProps> {
+  constructor(props: IParameterProps) {
+    super(props);
+  }
+
+  private addOption = (): void => {
+    const { options, updateParameterField } = this.props;
+    const newOptions = options.splice(0);
+    newOptions.push({ value: '' });
+    updateParameterField({ options: newOptions });
```

Comment:
```suggestion
    updateParameterField({ options: options.concat({ value: '' }) });
```
`concat` returns a new array, so this might be a little cleaner and save a few lines of code

Reply:
👍